runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -513,25 +513,45 @@ function LanguageTree:register_cbs(cbs)
   end
 end
 
+---Determine whether {inner} is completely inside {outer} (inner=outer -> true).
+---@param outer range
+---@param inner range
+---@return boolean
 ---@private
-local function tree_contains(tree, range)
-  local start_row, start_col, end_row, end_col = tree:root():range()
-  local start_fits = start_row < range[1] or (start_row == range[1] and start_col <= range[2])
-  local end_fits = end_row > range[3] or (end_row == range[3] and end_col >= range[4])
+local function range_contains(outer, inner)
+  local start_row, start_col, end_row, end_col = unpack(outer)
+  local start_fits = start_row < inner[1] or (start_row == inner[1] and start_col <= inner[2])
+  local end_fits = end_row > inner[3] or (end_row == inner[3] and end_col >= inner[4])
 
-  return start_fits and end_fits
+  if start_fits and end_fits then
+    return true
+  end
+
+  return false
 end
 
 --- Determines whether {range} is contained in this language tree
 ---
 ---@param range A range, that is a `{ start_line, start_col, end_line, end_col }` table.
 function LanguageTree:contains(range)
-  for _, tree in pairs(self._trees) do
-    if tree_contains(tree, range) then
-      return true
+  -- Insp(range)
+  for _, region in ipairs(self._regions) do
+    for _, node_range in ipairs(region) do
+      -- node_range is either node (userdata) or six-tuple as set in
+      -- |LanguageTree:set_included_regions|.
+      local region_range
+      if type(node_range) == "table" then
+        -- omit byte-offset of start/end-position.
+        region_range = {node_range[1], node_range[2], node_range[4], node_range[5]}
```

Comment:
Really: 1,2,4,5 ? Is the byte post stored at 3? Usually most ranges are srow, scol,erow,ecol. Possible that it's different in this file for `offset!`

Reply:
Yes, it's `{start_row, start_col, start_byte, end_row, end_col, end_byte}`

Reply:
(`region` != `range`)

Reply:
@L3MON4D3 I'd make sure you are using names here that are consistent with the rest of the codebase.

Reply:
True, they aren't that great. I would've just used the one from `set_included_regions` but `range` was already taken.
Are they okay now? Alternatively, I could change `range` to something like `query_range` and use the names from the loop in `LT:set_included_regions`

Reply:
Why use different structures at all and not just work with `range` or `region`, depending on what data you need?

Reply:
(But not with `stylua`; make sure to run `stylua runtime/` from the repo root.)

Reply:
I don't really get what you're suggesting, `contains` has to iterate over all regions that are part of this `LanguageTree` and find the one (if any) that includes `range`.
It's kind of unfortunate that `_regions` contains either a `node` or a `range`, but I don't know if that should be changed.

Reply:
I think "region" should be ok as it is the data structure applied to `set_included_regions`

Reply:
It's just about the naming. I think it's much better now; let's see if someone else has a better suggestion.

Reply:
@L3MON4D3  `_regions` actually contains the regions per tree on the first nesting level and the actual regions (which are ranges as a 6-tupel). So @clason 's idea about using a different name are justified. We could use `region_contains` or do a test for 4 or 6 tupels.

Reply:
I'd be fine with `region_contains` instead of `range_contains`, whichever you prefer :)
(that routine is only used in `contains`, so no big difference I'd say)

Reply:
Nvm that, it's much easier to reason about with `region_contains` :+1: